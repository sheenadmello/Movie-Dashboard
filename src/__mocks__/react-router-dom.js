// src/__mocks__/react-router-dom.js
import React from 'react';

export const BrowserRouter = ({ children }) => <div>{children}</div>;
export const Route = ({ children }) => <div>{children}</div>;
export const Link = ({ children }) => <div>{children}</div>;
export const useParams = () => ({ id: '1' });
