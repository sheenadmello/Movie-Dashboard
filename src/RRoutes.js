import React, { useState } from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom'; // Wrap with BrowserRouter
import MovieList from './components/MovieList';
import MovieDetails from './components/MovieDetails';
import SearchBar from './components/SearchBar';
import SortAndFilter from './components/SortAndFilter';
import ErrorBoundary from './components/ErrorBoundary';
import { StyledContainer, StyledCard } from './components/StyledComponents';

function RRoutes() {
  const [searchQuery, setSearchQuery] = useState('');

  return (
    <BrowserRouter> {/* Ensure everything is wrapped in BrowserRouter */}
      <ErrorBoundary>
        <StyledContainer>
          
            {/* These will be displayed only on the / route */}
            <Routes>
              <Route path="/" element={<><SearchBar setSearchQuery={setSearchQuery} /><SortAndFilter /><MovieList searchQuery={searchQuery} /></>} /> 
              <Route path="/movie/:id" element={<MovieDetails />} />
            </Routes>
          
        </StyledContainer>
      </ErrorBoundary>
    </BrowserRouter>
  );
}

export default RRoutes;

// import React, { useState } from 'react';
// import { BrowserRouter, Routes, Route } from 'react-router-dom';
// import MovieList from './components/MovieList';
// import MovieDetails from './components/MovieDetails';
// import SearchBar from './components/SearchBar';
// import SortAndFilter from './components/SortAndFilter';
// import ErrorBoundary from './components/ErrorBoundary';
// import { StyledContainer, StyledCard, StyledPaginationContainer, StyledButton } from './components/StyledComponents';

// function RRoutes() {
//   const [searchQuery, setSearchQuery] = useState('');

//   return (
//     <BrowserRouter>
//       <ErrorBoundary>
//         <StyledContainer>
//           <StyledCard>
//             <SearchBar setSearchQuery={setSearchQuery} />
//             <SortAndFilter />
//           </StyledCard>
//           <Routes>
//             <Route path="/" element={<MovieList searchQuery={searchQuery} />} />
//             <Route path="/movie/:id" element={<MovieDetails />} />
//           </Routes>

//         </StyledContainer>
//       </ErrorBoundary>
//     </BrowserRouter>
//   );
// }

// export default RRoutes;

