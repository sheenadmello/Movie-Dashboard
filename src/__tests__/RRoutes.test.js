import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import RRoutes from '../RRoutes'; 

// Create a mock reducer for the Redux store
const mockReducer = (state = { movie: {} }, action) => state;

// Create a mock store
const store = createStore(mockReducer);

// Create a QueryClient instance
const queryClient = new QueryClient();

describe('RRoutes Component', () => {
  test('renders without crashing', () => {
    const { container } = render(
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          <BrowserRouter>
            <RRoutes />
          </BrowserRouter>
        </QueryClientProvider>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});



