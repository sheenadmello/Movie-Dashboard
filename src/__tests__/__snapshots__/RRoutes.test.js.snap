// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RRoutes Component renders without crashing 1`] = `
<div>
  <div>
    <div>
       
      <div>
        Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.

Check the render method of \`RRoutes\`.
      </div>
    </div>
  </div>
</div>
`;
